src/swap/utils/processSwapTransaction.ts

```diff
@@ -1,56 +1,46 @@
 import type { Address } from 'viem';
 import { encodeFunctionData, parseAbi } from 'viem';
-import { waitForTransactionReceipt } from 'wagmi/actions';
+import { base } from 'viem/chains';
+import type { Call } from '../../transaction/types';
 import {
   PERMIT2_CONTRACT_ADDRESS,
   UNIVERSALROUTER_CONTRACT_ADDRESS,
 } from '../constants';
 import type { ProcessSwapTransactionParams } from '../types';
+import { sendSwapTransactions } from './sendSwapTransactions';
 
 export async function processSwapTransaction({
+  chainId,
   config,
+  sendCallsAsync,
   sendTransactionAsync,
   updateLifecycleStatus,
   swapTransaction,
+  switchChainAsync,
   useAggregator,
+  walletCapabilities,
 }: ProcessSwapTransactionParams) {
   const { transaction, approveTransaction, quote } = swapTransaction;
+  const transactions: Call[] = [];
 
   // for swaps from ERC-20 tokens,
   // if there is an approveTransaction present,
   // request approval for the amount
   // for V1 API, `approveTx` will be an ERC-20 approval against the Router
   // for V2 API, `approveTx` will be an ERC-20 approval against the `Permit2` contract
   if (approveTransaction?.data) {
-    updateLifecycleStatus({
```

Comment:
We lost all the `transactionPending`, what's the thinking behind this decision?

Reply:
we still have `transactionPending` - we have moved this to the `sendSwapTransactions` function

`processSwapTransaction` is now only responsible for building the `transactions` array